language/move-vm/runtime/src/interpreter.rs

```diff
@@ -1082,6 +1114,14 @@ impl Frame {
                             interpreter.operand_stack.last_n(field_count as usize)?,
                         )?;
                         let args = interpreter.operand_stack.popn(field_count)?;
+
+                        if self.paranoid_checks {
```

Comment:
Wondering whether we should have a different flag, so if this starts to produce false positives (the risk here higher as with type checks), we don't have to turn everthing off.

Reply:
Added a different flag for this.